datanymizer_engine/src/transformer/uniq_transformer.rs

```diff
@@ -0,0 +1,76 @@
+use super::{Globals, TransformResult, TransformResultHelper, Transformer, Uniqueness};
+use crate::uniq_collector;
+
+pub trait UniqTransformer {
+    // Or maybe we should move this to `Transformer` trait?
+    fn do_transform(
+        &self,
+        field_name: &str,
+        field_value: &str,
+        globals: &Option<Globals>,
+    ) -> String;
+
+    fn transform_with_retry(
+        &self,
+        field_name: &str,
+        field_value: &str,
+        globals: &Option<Globals>,
+    ) -> Option<String> {
+        let mut count = self.try_count();
+        while count > 0 {
+            let val = self.do_transform(field_name, field_value, globals);
+            if uniq_collector::add_to_collector(&field_name, &val) {
+                return Some(val);
+            } else {
+                count -= 1;
+            }
+        }
+        None
+    }
+
+    fn uniq(&self) -> &Uniqueness;
+
+    fn try_count(&self) -> i64 {
+        self.uniq()
+            .try_count
+            .unwrap_or_else(|| self.default_try_count())
+    }
+
+    fn default_try_count(&self) -> i64 {
+        // Or maybe 1?
```

Comment:
Maybe 1. Like "without retry".

Reply:
I think that in most cases there will be many available unique values, and errors due to random coincidence will be annoying. So I'll leave it at 3 for now (this will be very easy to change)